src/main/resources/hazelcast-config-schema.json

```diff
@@ -1373,20 +1373,535 @@
+              "type": "integer",
+              "default": 128,
+              "description": "Specifies the size of the receive buffer."
+            },
+            "linger-seconds": {
+              "type": "integer",
+              "default": 0,
+              "description": "Specifies the TCP linger seconds."
+            }
+          }
+        }
+      }
+    },
+    "SocketEndpointConfig": {
+      "$ref": "#/definitions/WanEndpointConfig",
+      "properties": {
+        "port": {
+          "type": "object",
+          "description": "The ports which Hazelcast will use to communicate between cluster members.",
+          "additionalProperties": false,
+          "properties": {
+            "port": {
+              "type": "integer",
+              "default": 5701
+            },
+            "port-count": {
+              "type": "integer",
+              "default": 100,
+              "description": "The default value is 100, meaning that Hazelcast will try to bind 100 ports. If you set the value of port as 5701, as members join the cluster, Hazelcast tries to find ports between 5701 and 5801. You can change the port count in cases like having large instances on a single machine or you are willing to have only a few ports assigned."
+            },
+            "auto-increment": {
+              "type": "boolean",
+              "default": true,
+              "description": "If port is set to 5701, Hazelcast will try to find free ports between 5701 and 5801. Normally, you will not need to change this value, but it comes in handy when needed. You may also want to choose to use only one port. In that case, you can disable the auto-increment feature of port by setting its value as false."
+            }
+          }
+        },
+        "public-address": {
+          "type": "string",
+          "description": "Overrides the public address of a node. By default, a node selects its socket address as its public address. But behind a network address translation (NAT), two endpoints (nodes) may not be able to see/access each other. If both nodes set their public addresses to their defined addresses on NAT, then they can communicate with each other. In this case, their public addresses are not an address of a local network interface but a virtual address defined by NAT. This is optional to set and useful when you have a private cloud."
+        },
+        "reuse-address": {
+          "type": "boolean",
+          "default": false,
+          "description": "When you shutdown a cluster member, the server socket port will be in the TIME_WAIT state for the next couple of minutes. If you start the member right after shutting it down, you may not be able to bind it to the same port because it is in the TIME_WAIT state. If you set reuse-address to true, the TIME_WAIT state is ignored and you can bind the member to the same port again."
+        }
+      }
+    },
+    "AdvancedNetwork": {
+      "additionalProperties": false,
```

Comment:
Please add "type": "object" too. Otherwise, if the actual input is not an object, then all property validations will be ignored, and it will be a false success (silent failure)
.

Reply:
done